foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/expressions/ExpressionOperator.java

```diff
@@ -2192,7 +2194,9 @@ public void printCollection(Vector items, ExpressionSQLPrinter printer) {
                 printer.printString(alias.getNameDelimited(printer.getPlatform()));
             } else if ((this.selector == Count) && (item.isExpressionBuilder())) {
                 printer.printString("*");
-            } else {
+            } else if ((this.selector == Case) && (item.isConstantExpression())) {

```

Comment:
I dont think this is the solution. Both the THEN and CLAUSE items are ConstantExpressions. We want to call ```item.printSQL(printer)``` for the THEN clauses so that they use the baseExpression.